MongoDB/todoApp.js

```diff
@@ -12,98 +12,108 @@ app.use(bodyParser.urlencoded({
-    db.collection('work').find(query).toArray(function(err, docs){
-        res.send(docs);
-      });
-});
-
-//delete
-app.delete('/task/:id', function(req, res) {
+var getDocuments = function(req, res, urlx) {
+  if(isNaN(parseInt(req.params.id)) == true){
+    res.send('Wrong Input!!! (Maybe it is not number)', 400);
+  }
+  else{
   var myquery = { ID: parseInt(req.params.id)};
-  db.collection('task').remove(myquery, function(err, docs) {
-    db.collection('task').find(myquery).toArray(function(err, docs){
-        res.send(docs);
-    });
+  db.collection(urlx).findOne(myquery,function(err, docs){
+    if(docs == null){
+      res.send('Not found :-(', 404);
+    } else {
+      res.send(docs);}
   });
-});
-app.delete('/work/:id', function(req, res) {
+  }
+}
+
+//delete
+var deleteDocuments = function(req, res, urlx) {
+  if(isNaN(parseInt(req.params.id)) == true){
+    res.send('Wrong Input!!! (Maybe it is not number)', 400);
+  }
+  else{
   var myquery = { ID: parseInt(req.params.id)};
-  db.collection('work').remove(myquery, function(err, docs) {
-    db.collection('work').find(myquery).toArray(function(err, docs){
-        res.send(docs);
+  db.collection(urlx).remove(myquery, function(err, docs){
+    db.collection(urlx).findOne(myquery,function(err, docs){
+        res.send(docs,204);
     });
   });
-});
+  }
+}
 
 //update
-app.patch('/task/:id/:description', function(req, res){
-  var myquery = { ID: parseInt(req.params.id)};
-  var newquery = {$set: {Description: req.params.description}}
-  db.collection('task').findOneAndUpdate(myquery,newquery,{upsert: true},function(err,docs){
-      db.collection('task').find(myquery).toArray(function(err, docs){
-          res.send(docs);
-      });
-  });
-});
-app.patch('/work/:id/:description', function(req, res){
+var updateDocuments = function(req, res, urlx) {
+  var json = req.body;
+  if(isNaN(parseInt(req.params.id)) == true){
+    res.send('Wrong Input!!! (Maybe it is not number)', 400);
+  } else {
   var myquery = { ID: parseInt(req.params.id)};
-  var newquery = {$set: {Description: req.params.description}}
-  db.collection('work').findOneAndUpdate(myquery,newquery,{upsert: true},function(err,docs){
-      db.collection('work').find(myquery).toArray(function(err, docs){
-          res.send(docs);
-      });
-  });
-});
+  var newquery = {$set: {Description: json.Description,
+                        Completed: json.Completed,
+                        Duration: json.Duration}}
+    if(typeof json.Description !== 'string'||typeof json.Completed !== 'boolean'||typeof json.Duration !== 'string'){
+      res.send('Something wrong???', 400);}
+    else{
+          db.collection(urlx).findOneAndUpdate(myquery,newquery,function(err,docs){
+            db.collection(urlx).findOne(myquery,function(err, docs){
+              res.send(docs);
+            });
+          });
+        }
+    }
+}
+
+//insert
+app.post('/task', function(req, res){insertDocuments(req, res, 'task');});
```

Comment:
Although (req, res) works fine, you should keep the middleware header as (req, res, next) to allow future control flow.